proto/interchain_security/ccv/provider/v1/provider.proto

```diff
@@ -196,9 +191,57 @@ message VscSendTimestamp {
   google.protobuf.Timestamp timestamp = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
 }
 
+//
+// Key assignment section
+//
+
+// A validator's assigned consensus address for a consumer chain. 
+// Note this type is for type safety within provider code, consumer code uses normal sdk.ConsAddress, 
+// since there's no notion of provider vs consumer address.
+message ConsumerConsAddress {
+  // Do not generate stringer for this type, we'll use sdk.ConsAddress.String() instead
+  option (gogoproto.goproto_stringer) = false;
+  bytes address = 1;
+}
+
+// A validator's consensus address on the provider chain
+message ProviderConsAddress {
+  // Do not generate stringer for this type, we'll use sdk.ConsAddress.String() instead
+  option (gogoproto.goproto_stringer) = false;
+  bytes address = 1;
+}
+
+// AddressList contains a list of consumer consensus addresses
+message ConsumerAddressList {
+  repeated ConsumerConsAddress addresses = 1;
+}
+  
 message KeyAssignmentReplacement {
-  bytes provider_addr = 1;
+  ProviderConsAddress provider_addr = 1;
```

Comment:
The CLI fills the `KeyAssignmentReplacement` msg using the validator operator address
see how we it's parsed [here](https://github.com/cosmos/interchain-security/blob/bf6cdae5f02ff31bb2982766bed9db4c664a0ec3/x/ccv/provider/keeper/msg_server.go#L38)